File: samples/client/3_1_0_unit_test/java/src/test/java/org/openapijsonschematools/client/components/schemas/SmallMultipleOfLargeIntegerTest.java
package org.openapijsonschematools.client.components.schemas;

import org.junit.Assert;
import org.junit.Test;
import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags;
import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;
import org.openapijsonschematools.client.schemas.validation.MapUtils;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.AbstractMap;

public class SmallMultipleOfLargeIntegerTest {
    static final SchemaConfiguration configuration = new SchemaConfiguration(JsonSchemaKeywordFlags.onlyFormat());

    @Test
    public void testAnyIntegerIsAMultipleOf1E8Passes() throws ValidationException {
        // any integer is a multiple of 1e-8
        final var schema = SmallMultipleOfLargeInteger.SmallMultipleOfLargeInteger1.getInstance();
        schema.validate(
            12391239123L,
            configuration
        );
    }
}

File: samples/client/3_1_0_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/components/schemas/SmallMultipleOfLargeIntegerTest.kt
package org.openapijsonschematools.client.components.schemas

import org.openapijsonschematools.client.configurations.JsonSchemaKeyword
import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

import kotlin.test.Test
import kotlin.test.assertFailsWith

class SmallMultipleOfLargeIntegerTest {
    companion object {
        val configuration = SchemaConfiguration(disabledKeywordFlags=setOf(JsonSchemaKeyword.FORMAT))
    }

    @Test
    fun testAnyIntegerIsAMultipleOf1E8Passes() {
        // any integer is a multiple of 1e-8
        val schema = SmallMultipleOfLargeInteger.SmallMultipleOfLargeInteger1.getInstance()
        schema.validate(
            12391239123L,
            configuration
        )
    }
}
